app/crud/crud_project.py

```diff
@@ -61,16 +62,63 @@ def archive(self, db: Session, db_obj: Project) -> Project:
     def unarchive(self, db: Session, db_obj: Project) -> Project:
         return self.update(db=db, db_obj=db_obj, obj_in={"archived_at": None})
 
+    def __validate_model(self, obj_in: dict) -> bool:
+        func_name = "validate_model"
+        func_validate_model = getattr(program_class, func_name, None)
+
+        if not func_validate_model:
+            raise AttributeError(f"{func_name} is not a method of {self}")
+        if not callable(func_validate_model):
+            raise TypeError(f"{func_name} is not callable")
+
+        return func_validate_model(program_class, jsonable_encoder(obj_in))
```

Comment:
```suggestion
        return func_validate_model(program_class, obj_in)
```
`jsonable_encoder` is already used before calling this method